d2/src/main/java/com/linkedin/d2/balancer/simple/ClusterLoadBalancerSubscriber.java

```diff
@@ -105,20 +105,20 @@ protected void handleRemove(final String listenTo)
    */
   private ActivePropertiesResult pickActiveProperties(final ClusterProperties discoveryProperties)
   {
-    final ClusterStoreProperties clusterStoreProperties = toClusterStoreProperties(discoveryProperties);
-    if (clusterStoreProperties == null)
-    {
-      // this should not happen since the serializer returns the composite class
-      return discoveryProperties;
-    }
-
+    ClusterProperties pickedProperties = discoveryProperties;
```

Comment:
Merging the master branch causes changes from other commits also showing up in your changes, which is very confusing. Please make sure to use **git rebase** instead of merge, referencing: https://iwww.corp.linkedin.com/wiki/cf/pages/viewpage.action?spaceKey=ENGS&title=Pegasus+Git+Workflow#PegasusGitWorkflow-Merging

Reply:
TIL. Will do that next time.